fixups/RegLegacyFixups/RegistryFixups.cpp

```diff
@@ -257,6 +261,28 @@ REGSAM RegFixupSam(std::string keypath, REGSAM samDesired, DWORD RegLocalInstanc
     return samModified;
 }
 
+template <typename CharT>
+const char* ReplaceRegistryQueryPath(PHKEY key, const CharT* subKey)
+{
+    std::string keypath = InterpretKeyPath(*key) + "\\" + InterpretStringA(subKey);
```

Comment:
crash if key is `nullptr`

Reply:
How? We are always calling this function with the pointer to HKEY we got from the main application as argument. `key` over here will always point to the argument passed on to us and never be null.

Reply:
Just because you're calling this now with a non-null PHKEY does not mean, in the future this will always be called with a non-null PHKEY.

Additionally, this is OSS.  Other people can fork this and make their own and they might not know that PHKEY should be not-null.

Additionally, checking for a non-null pointer is good practice against crashing.

Reply:
Still not convinced we should add a a check even though all code paths guarantee non null parameter.  This is an irrecoverable error and we should crash if called with null PHKEY since we cannot read or write the underlying HKEY.

Added required annotations to the parameters so that compiler can raise warnings if it finds incorrect usage of the function. It will also indicate the assumptions made by the function explicit in the signature itself.